api/utils/keys/privatekey.go

```diff
@@ -214,14 +214,20 @@ func LoadPrivateKey(keyFile string) (*PrivateKey, error) {
 
 // ParsePrivateKey returns the PrivateKey for the given key PEM block.
 func ParsePrivateKey(keyPEM []byte) (*PrivateKey, error) {
+	return ParsePrivateKeyWithCustomPrompt(keyPEM, nil)
+}
+
+// ParsePrivateKeyWithCustomPrompt returns the PrivateKey for the given key PEM block.
+// Allows passing a custom hardware key prompt.
+func ParsePrivateKeyWithCustomPrompt(keyPEM []byte, customPrompt HardwareKeyPrompt) (*PrivateKey, error) {
```

Comment:
I don't really like this duality :(
Unfortunately, `ParsePrivateKey` is used in so many places, that updating all of them is too cumbersome. Additionally, I don't want to force the callsites to pass `nil` to this function when they don't really care about the `customPrompt`.

Reply:
This is really cool, applied now :D